torch/csrc/jit/function.h

```diff
@@ -39,6 +41,15 @@ struct TORCH_API Function {
     return graph_;
   }
 
+  std::shared_ptr<Graph> optimized_graph() const {
+    if (!optimized_graph_) {
```

Comment:
This isn't thread safe. Functions are invokable from multiple threads (e.g. in Predictor).